### PR TITLE
feat: separate workflow for unit tests

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- separate unit tests from build
- remove running unit tests from Dockerfiles

**Related issue(s)**

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [ ] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

File: .github/workflows/pull-unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: setup_busola
+        shell: bash
+        run: |
+          set -e
+          npm ci
```
Comment (dbadura) [inline .github/workflows/pull-unit-tests.yml:33]: We can use alias: `clean-install` to be easier to understand what this step is doing.

**Review by dbadura — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    types: [opened, edited, synchronize, reopened, ready_for_review]
+    paths:
+      - ".github/workflows/pull-unit-tests.yml"
+      - "backend/**"
+      - "public/**"
+      - "src/**"
+      - "tests/**"
+      - "package.json"
+      - "Makefile"
+      - "Dockerfile.local"
```
Comment (dbadura) [inline .github/workflows/pull-unit-tests.yml:14]: Do we need to run test on Dockerfile changes?

File: .github/workflows/pull-unit-tests.yml (hunk 1 above)
Comment (mrCherry97) [reply .github/workflows/pull-unit-tests.yml:33]: we are using `npm ci` everywhere and it's clear, so I'll go with that to keep the convention on which we agreed, if we want to change that it should be changed in all of the workflows.

Comment (dbadura) [reply .github/workflows/pull-unit-tests.yml:29]: We can extrend `ci` to `clean-install`, looks ok

**Review by OliwiaGowor — APPROVED** [review]

**Pull request merged**